### PR TITLE
allow local urls

### PR DESCRIPTION
**Pull request by casz**
does now allow for .local or machine/ 😓

### CODE REVIEW

File: plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/PluginParser.java [hunk 1]
```diff
@@ -88,7 +90,7 @@
                         String version = versionObject == null ? "latest" : versionObject.toString();
                         Object urlObject = pluginSource.get("url");
                         String url;
-                        if (urlObject != null && urlValidator.isValid(urlObject.toString())) {
+                        if (urlObject != null && URL_VALIDATOR.isValid(urlObject.toString())) {
                             url = urlObject.toString();
                         } else {
                             url = null;
```
Comment (casz) [inline plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/PluginParser.java]: perhaps a good idea to log invalid URLs?

**Comment by casz** [comment]
Hmm for some reason it doesn't think my urls are valid: `gitlab-branch-source::https://artifactorydk.3shape.local/maven/argelbargel/jenkins/plugins/gitlab-branch-source/DEV-SNAPSHOT/gitlab-branch-source-DEV-20190711.100116-2.hpi`

**Comment by timja** [comment]
> Hmm for some reason it doesn't think my urls are valid: `gitlab-branch-source::https://artifactorydk.3shape.local/maven/argelbargel/jenkins/plugins/gitlab-branch-source/DEV-SNAPSHOT/gitlab-branch-source-DEV-20190711.100116-2.hpi`

add a test?

**Comment by casz** [comment]
Ya, I did, but it's a bug I would say in validator we use...
[Local TLD's is `localdomain` and `localhost`](https://github.com/apache/bval/blob/e0fe2cd9754abd730086cf893407cc73c3690eac/bval-extras/src/main/java/org/apache/bval/extras/constraints/net/DomainValidator.java#L444-L446)

**Comment by casz** [comment]
Can we change the URL validator to something that isn't as strict?

**Comment by timja** [comment]
> Can we change the URL validator to something that is not as strict?

Go for it

File: plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/PluginParser.java (hunk 1 above)
Comment (kwhetstone) [reply plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/PluginParser.java]: That would be a good idea

File: plugin-management-cli/src/test/java/io/jenkins/tools/pluginmanager/cli/URLValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            "ftp://jenkins.io"
+        };
+    }
+
+    public URLValidatorTest(String url) {
+        this.url = url;
+    }
+
+    @Test
+    public void validURLs() {
```
Comment (timja) [inline plugin-management-cli/src/test/java/io/jenkins/tools/pluginmanager/cli/URLValidatorTest.java]: how about testing an invalid url or two?

Comment (kwhetstone) [reply plugin-management-cli/src/test/java/io/jenkins/tools/pluginmanager/cli/URLValidatorTest.java]: tests ftw

Comment (casz) [reply plugin-management-cli/src/test/java/io/jenkins/tools/pluginmanager/cli/URLValidatorTest.java]: :swea

**Pull request merged**